neocore/Cryptography/ECCurve.py

```diff
@@ -589,10 +626,15 @@ def decode_from_reader(self, reader):
         elif f == 4 or f == 6 or f == 7:
             raise NotImplementedError()
 
-        raise Exception("Invalid point incoding: %s " % f)
+        raise TypeError(f"Invalid point incoding: {f}")
 
     def decode_from_hex(self, hex_str, unhex=True):
-
+        """
+        Raises:
+            ValueError: if the hex_str is an incorrect length for encoding or compressed encoding
+            NotImplementedError: if the read int `f` == 6, or 7
```

Comment:
```suggestion
            NotImplementedError: if an unsupported point encoding is used
```
When writing docstrings try to write them from the perspective of _not_ knowing the implementation of the function. As a reader/user all you see is basically:
```
    def decode_from_hex(self, hex_str, unhex=True):

         """
        Raises:
            ValueError: if the hex_str is an incorrect length for encoding or compressed encoding
            NotImplementedError: if the read int `f` == 6, or 7
```
So without knowing the code a description like `if the read int `f` == 6, or 7` will not help the reader.

Reply:
Okay, this makes a lot of sense. Thank you for the feedback!